package casock.rpc.protobuf.api;

enum ResponseType {
  RESPONSE_TYPE_OK = 1;
  RESPONSE_TYPE_ERROR = 2;
};

message RpcRequest {
  required uint32 id = 1;
  required string operation = 2;
  optional bytes request = 3;
}

message RpcResponse {
  required uint32 id = 1;
  required ResponseType type = 2;
  optional bytes response = 3;
}
